main/lsp/requests/completion.cc

```diff
@@ -262,8 +277,52 @@ unique_ptr<CompletionItem> getCompletionItemForKeyword(const core::GlobalState &
     return item;
 }
 
+unique_ptr<CompletionItem> getCompletionItemForLocal(const core::GlobalState &gs, const LSPConfiguration &config,
+                                                     const core::LocalVariable &local, const core::Loc queryLoc,
+                                                     string_view prefix, size_t sortIdx) {
+    auto label = string(local._name.data(gs)->shortName(gs));
+    auto item = make_unique<CompletionItem>(label);
+    item->sortText = fmt::format("{:06d}", sortIdx);
+    item->kind = CompletionItemKind::Variable;
+
+    // TODO(jez) This should probably be a helper function (see getCompletionItemForSymbol)
+    u4 queryStart = queryLoc.beginPos();
+    u4 prefixSize = prefix.size();
+    auto replacementLoc = core::Loc{queryLoc.file(), queryStart - prefixSize, queryStart};
+    auto replacementRange = Range::fromLoc(gs, replacementLoc);
+    auto replacementText = label;
+    if (replacementRange != nullptr) {
+        item->textEdit = make_unique<TextEdit>(std::move(replacementRange), replacementText);
+    } else {
+        // TODO(jez) Why is replacementRange nullptr? instrument this and investigate when it fails
```

Comment:
I think this happens if `!replacementLoc.exists()` -- since then we can't get a filename out of it.